### PR TITLE
Add Homepage Markdown Plugin

### PR DESCRIPTION
**Pull request by fogleman**
@zachmullen @mgrauer 

This plugin allows an admin to replace the girder homepage with markdown-formatted content. Currently, image uploads are not handled in the markdown editor.

### CODE REVIEW

File: plugins/homepage/plugin_tests/homepage_test.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2014 Kitware Inc.
```
Comment (mgrauer) [inline plugins/homepage/plugin_tests/homepage_test.py]: `Copyright 2016 Kitware Inc.`

Although, it is kinda goofy that we have copyright on Python files and no others.  Not sure why that is.

File: plugins/homepage/server/__init__.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
```
Comment (mgrauer) [inline plugins/homepage/server/__init__.py]: `Copyright 2016 Kitware Inc.`

**Comment by mgrauer** [comment]
I have a mild preference to call this landing page instead of home page, but currently not enough to ask you to rename anything.  If no one chimes in before the PR is done, it can stay.

**Comment by zachmullen** [comment]
The overridden view is called "Front Page" in the code, which is another option for the name.

**Comment by mgrauer** [comment]
- [ ] Add a brief README explaining what this does with a screenshot, the screenshot should show a before and after of the Girder front page with markdown

**Comment by cpatrick** [comment]
Sphinx docs would be preferable to a readme.

**Comment by mgrauer** [comment]
LGTM.

🔫 

@fogleman  is killing it.

**Pull request merged**